### PR TITLE
Name MusicXML files, types and methods consistently

### PR DESCRIPTION
**Pull request by miiizen**
This PR clears up the varied use of `musicXML`, `musicXml`, `mxml` etc. and replaces them all with `MusicXML` (or `musicXML` if at the start of a method name).
This affects all names except for private variable names and parameters.

### CODE REVIEW

**Review by cbjeukendrup — COMMENTED** [review]
I hate to nit-pick on everything, but note that in existing code we've usually written `Xml` (as in `XmlStreamReader`), and also `Json` rather than `JSON`, `Midi` rather than `MIDI`. So for the sake of consistency, one could argue that `MusicXml` is preferable. But I won't insist on this; if you prefer `MusicXML` that's also fine.

**Comment by miiizen** [comment]
No - you're right, please continue nitpicking!  Sorted.

File: src/importexport/musicxml/internal/musicxml/exportmusicxml.cpp [hunk 1]
```diff
@@ -21,11 +21,11 @@
  */
 
 /**
- MusicXML export.
+ MusicXml export.
  */
 
 // TODO: trill lines need to be handled the same way as slurs
-// in MuseScore they are measure level elements, while in MusicXML
+// in MuseScore they are measure level elements, while in MusicXml
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/exportmusicxml.cpp:28]: Hm, maybe this change should not apply to comments / natural language. In natural languages, we should of course just use the official spelling, i.e. JSON, MIDI, MusicXML, but only in CamelCase that becomes `Json`, `Midi`, `MusicXml`,

File: src/importexport/musicxml/imusicxmlconfiguration.h [hunk 2]
```diff
@@ -32,27 +32,27 @@ class IMusicXmlConfiguration : MODULE_EXPORT_INTERFACE
 public:
     virtual ~IMusicXmlConfiguration() = default;
 
-    virtual bool musicxmlImportBreaks() const = 0;
-    virtual void setMusicxmlImportBreaks(bool value) = 0;
+    virtual bool musicXmlImportBreaks() const = 0;
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/imusicxmlconfiguration.h:35]: Maybe we can just drop these `musicXml` prefixes here; it's clear enough that these things are about MusicXML :)

**Review by cbjeukendrup — APPROVED** [review]

**Pull request merged**